In [2]:
import matplotlib.pyplot as plt
import numpy as np
import seaborn
import pandas as pd
from pandas import DataFrame, Series
%matplotlib inline

In [3]:
batting_data = pd.read_csv('Data/core/Batting.csv', sep = ",", dtype = str).fillna(0)

In [4]:
batting_data = batting_data[['playerID', 'yearID', 'H', 'BB', 'HBP', 'AB', 'SF']]

In [5]:
batting_data.head()

playerID yearID   H BB HBP   AB SF
0  abercda01   1871   0  0   0    4  0
1   addybo01   1871  32  4   0  118  0
2  allisar01   1871  40  2   0  137  0
3  allisdo01   1871  44  0   0  133  0
4  ansonca01   1871  39  2   0  120  0

In [6]:
salary_data = pd.read_csv('Data/core/Salaries.csv', sep = ",", dtype = str).fillna(0)

In [7]:
salary_data = salary_data[['playerID', 'salary']]

In [8]:
salary_data.head()

playerID  salary
0  barkele01  870000
1  bedrost01  550000
2  benedbr01  545000
3   campri01  633333
4  ceronri01  625000

In [9]:
fielding_data = pd.read_csv('Data/core/Fielding.csv', sep = ",", dtype = str).fillna(0)

In [10]:
fielding_data = fielding_data[['playerID', 'POS']]

In [11]:
fielding_data.head()

playerID POS
0  abercda01  SS
1   addybo01  2B
2   addybo01  SS
3  allisar01  2B
4  allisar01  OF

In [12]:
batting_and_salary = pd.merge(batting_data, salary_data)

In [13]:
batting_and_salary.head()

playerID yearID   H BB HBP  AB SF  salary
0  lawrebr01   1954  10  3   1  53  0  500000
1  lawrebr01   1955   2  1   0  21  0  500000
2  lawrebr01   1956  11  2   1  70  2  500000
3  lawrebr01   1957  14  6   0  82  0  500000
4  lawrebr01   1958   6  5   0  53  2  500000

In [14]:
batting_and_salary.sort_values('playerID')

playerID yearID    H  BB HBP   AB SF   salary
258764  aardsda01   2013    0   0   0    0  0   403250
258746  aardsda01   2009    0   0   0    0  0  4500000
258747  aardsda01   2009    0   0   0    0  0   500000
258748  aardsda01   2010    0   0   0    0  0   300000
258749  aardsda01   2010    0   0   0    0  0   387500
258750  aardsda01   2010    0   0   0    0  0   403250
258751  aardsda01   2010    0   0   0    0  0   419000
258752  aardsda01   2010    0   0   0    0  0  2750000
258753  aardsda01   2010    0   0   0    0  0  4500000
258754  aardsda01   2010    0   0   0    0  0   500000
258757  aardsda01   2012    0   0   0    0  0   403250
258775  aardsda01   2015    0   0   0    1  0   500000
258774  aardsda01   2015    0   0   0    1  0  4500000
258773  aardsda01   2015    0   0   0    1  0  2750000
258745  aardsda01   2009    0   0   0    0  0  2750000
258772  aardsda01   2015    0   0   0    1  0   419000
258770  aardsda01   2015    0   0   0    1  0   387500
258769  aardsda01   2015    0   0   0    1  0   300000
258768  aardsda01   2013    0   0   0    0  0   500000
258767  aardsda01   2013    0   0   0    0  0  4500000
258766  aardsda01   2013    0   0   0    0  0  2750000
258765  aardsda01   2013    0   0   0    0  0   419000
258755  aardsda01   2012    0   0   0    0  0   300000
258763  aardsda01   2013    0   0   0    0  0   387500
258762  aardsda01   2013    0   0   0    0  0   300000
258761  aardsda01   2012    0   0   0    0  0   500000
258760  aardsda01   2012    0   0   0    0  0  4500000
258759  aardsda01   2012    0   0   0    0  0  2750000
258758  aardsda01   2012    0   0   0    0  0   419000
258771  aardsda01   2015    0   0   0    1  0   403250
...           ...    ...  ...  ..  ..  ... ..      ...
298936  zuninmi01   2013   37  16   3  173  1   504100
298937  zuninmi01   2013   37  16   3  173  1   523500
298938  zuninmi01   2014   87  17  17  438  4   504100
298940  zuninmi01   2015   61  21   5  350  2   504100
298941  zuninmi01   2015   61  21   5  350  2   523500
298939  zuninmi01   2014   87  17  17  438  4   523500
138846  zupcibo01   1993   69  27   2  286  3   222000
138851  zupcibo01   1994   18   4   0   88  1   109000
138850  zupcibo01   1994   18   4   0   88  1   100000
138849  zupcibo01   1994    0   0   0    4  0   222000
138848  zupcibo01   1994    0   0   0    4  0   109000
138847  zupcibo01   1994    0   0   0    4  0   100000
138844  zupcibo01   1993   69  27   2  286  3   100000
138843  zupcibo01   1992  108  25   4  392  4   222000
138845  zupcibo01   1993   69  27   2  286  3   109000
138838  zupcibo01   1991    4   1   0   25  0   100000
138839  zupcibo01   1991    4   1   0   25  0   109000
138852  zupcibo01   1994   18   4   0   88  1   222000
138841  zupcibo01   1992  108  25   4  392  4   100000
138842  zupcibo01   1992  108  25   4  392  4   109000
138840  zupcibo01   1991    4   1   0   25  0   222000
43980   zuvelpa01   1991    0   0   0    0  0   145000
43979   zuvelpa01   1989   16   1   1   58  0   145000
43977   zuvelpa01   1987    6   0   0   34  0   145000
43976   zuvelpa01   1986    4   5   0   48  0   145000
43975   zuvelpa01   1985   48  16   0  190  0   145000
43974   zuvelpa01   1984    5   2   0   25  0   145000
43973   zuvelpa01   1983    0   2   1    5  0   145000
43972   zuvelpa01   1982    0   0   0    1  0   145000
43978   zuvelpa01   1988   30   8   0  130  0   145000

[299134 rows x 8 columns]

In [15]:
amalgamated_data = pd.merge(batting_and_salary, fielding_data)

In [16]:
amalgamated_data.head()

playerID yearID   H BB HBP  AB SF  salary POS
0  lawrebr01   1954  10  3   1  53  0  500000   P
1  lawrebr01   1954  10  3   1  53  0  500000   P
2  lawrebr01   1954  10  3   1  53  0  500000   P
3  lawrebr01   1954  10  3   1  53  0  500000   P
4  lawrebr01   1954  10  3   1  53  0  500000   P

In [17]:
amalgamated_data.sort_values('playerID')

playerID yearID   H  BB HBP   AB SF  salary POS
7168248  aardsda01   2009   0   0   0    0  0  403250   P
7168172  aardsda01   2008   0   0   0    1  0  300000   P
7168171  aardsda01   2008   0   0   0    1  0  300000   P
7168170  aardsda01   2008   0   0   0    1  0  300000   P
7168169  aardsda01   2008   0   0   0    1  0  300000   P
7168168  aardsda01   2008   0   0   0    1  0  300000   P
7168167  aardsda01   2008   0   0   0    1  0  300000   P
7168173  aardsda01   2008   0   0   0    1  0  300000   P
7168166  aardsda01   2008   0   0   0    1  0  300000   P
7168164  aardsda01   2007   0   0   0    0  0  500000   P
7168163  aardsda01   2007   0   0   0    0  0  500000   P
7168162  aardsda01   2007   0   0   0    0  0  500000   P
7168161  aardsda01   2007   0   0   0    0  0  500000   P
7168160  aardsda01   2007   0   0   0    0  0  500000   P
7168159  aardsda01   2007   0   0   0    0  0  500000   P
7168165  aardsda01   2007   0   0   0    0  0  500000   P
7168158  aardsda01   2007   0   0   0    0  0  500000   P
7168174  aardsda01   2008   0   0   0    1  0  300000   P
7168176  aardsda01   2008   0   0   0    1  0  387500   P
7168190  aardsda01   2008   0   0   0    1  0  403250   P
7168189  aardsda01   2008   0   0   0    1  0  403250   P
7168188  aardsda01   2008   0   0   0    1  0  403250   P
7168187  aardsda01   2008   0   0   0    1  0  403250   P
7168186  aardsda01   2008   0   0   0    1  0  403250   P
7168185  aardsda01   2008   0   0   0    1  0  403250   P
7168175  aardsda01   2008   0   0   0    1  0  387500   P
7168184  aardsda01   2008   0   0   0    1  0  403250   P
7168182  aardsda01   2008   0   0   0    1  0  387500   P
7168181  aardsda01   2008   0   0   0    1  0  387500   P
7168180  aardsda01   2008   0   0   0    1  0  387500   P
...            ...    ...  ..  ..  ..  ... ..     ...  ..
1514704  zuvelpa01   1985  48  16   0  190  0  145000  SS
1514703  zuvelpa01   1985  48  16   0  190  0  145000  3B
1514702  zuvelpa01   1985  48  16   0  190  0  145000  2B
1514701  zuvelpa01   1985  48  16   0  190  0  145000  SS
1514700  zuvelpa01   1985  48  16   0  190  0  145000  SS
1514699  zuvelpa01   1985  48  16   0  190  0  145000  3B
1514698  zuvelpa01   1985  48  16   0  190  0  145000  2B
1514697  zuvelpa01   1985  48  16   0  190  0  145000  SS
1514696  zuvelpa01   1985  48  16   0  190  0  145000  2B
1514695  zuvelpa01   1985  48  16   0  190  0  145000  SS
1514694  zuvelpa01   1985  48  16   0  190  0  145000  SS
1514693  zuvelpa01   1984   5   2   0   25  0  145000  3B
1514692  zuvelpa01   1984   5   2   0   25  0  145000  SS
1514706  zuvelpa01   1985  48  16   0  190  0  145000  3B
1514691  zuvelpa01   1984   5   2   0   25  0  145000  DH
1514689  zuvelpa01   1984   5   2   0   25  0  145000  SS
1514688  zuvelpa01   1984   5   2   0   25  0  145000  SS
1514687  zuvelpa01   1984   5   2   0   25  0  145000  3B
1514686  zuvelpa01   1984   5   2   0   25  0  145000  2B
1514685  zuvelpa01   1984   5   2   0   25  0  145000  SS
1514684  zuvelpa01   1984   5   2   0   25  0  145000  SS
1514683  zuvelpa01   1984   5   2   0   25  0  145000  3B
1514682  zuvelpa01   1984   5   2   0   25  0  145000  2B
1514681  zuvelpa01   1984   5   2   0   25  0  145000  SS
1514680  zuvelpa01   1984   5   2   0   25  0  145000  2B
1514679  zuvelpa01   1984   5   2   0   25  0  145000  SS
1514678  zuvelpa01   1984   5   2   0   25  0  145000  SS
1514677  zuvelpa01   1983   0   2   1    5  0  145000  3B
1514690  zuvelpa01   1984   5   2   0   25  0  145000  3B
1514788  zuvelpa01   1991   0   0   0    0  0  145000  SS

[7781783 rows x 9 columns]

In [18]:
# relevant_data = amalgamated_data[['playerID', 'yearID', 'AB', 'H', 'BB', 'HBP', 'SF', 'salary', 'POS']]

In [19]:
# relevant_data.head()

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

In [20]:
float_data = amalgamated_data[['yearID', 'AB', 'H', 'BB', 'HBP', 'SF', 'salary']].astype(float)

In [21]:
amalgamated_data.pop('yearID')

0          1954
1          1954
2          1954
3          1954
4          1954
5          1954
6          1954
7          1955
8          1955
9          1955
10         1955
11         1955
12         1955
13         1955
14         1956
15         1956
16         1956
17         1956
18         1956
19         1956
20         1956
21         1957
22         1957
23         1957
24         1957
25         1957
26         1957
27         1957
28         1958
29         1958
           ... 
7781753    2015
7781754    2015
7781755    2015
7781756    2015
7781757    2015
7781758    2014
7781759    2014
7781760    2015
7781761    2015
7781762    2014
7781763    2014
7781764    2015
7781765    2015
7781766    2014
7781767    2014
7781768    2015
7781769    2015
7781770    2014
7781771    2014
7781772    2015
7781773    2015
7781774    2015
7781775    2015
7781776    2015
7781777    2015
7781778    2015
7781779    2015
7781780    2015
7781781    2015
7781782    2015
Name: yearID, dtype: obj

In [22]:
amalgamated_data.pop('salary')

0           500000
1           500000
2           500000
3           500000
4           500000
5           500000
6           500000
7           500000
8           500000
9           500000
10          500000
11          500000
12          500000
13          500000
14          500000
15          500000
16          500000
17          500000
18          500000
19          500000
20          500000
21          500000
22          500000
23          500000
24          500000
25          500000
26          500000
27          500000
28          500000
29          500000
            ...   
7781753     508700
7781754     508700
7781755     508700
7781756     508700
7781757     508700
7781758     512800
7781759     512800
7781760     512800
7781761     512800
7781762     517500
7781763     517500
7781764     517500
7781765     517500
7781766    4000000
7781767    4000000
7781768    4000000
7781769    4000000
7781770     512800
7781771     512800
7781772     512800
7781773     512800
7781774     

In [23]:
cleaned_data = pd.concat([amalgamated_data, float_data], axis = 1)

In [24]:
cleaned_data.head()

playerID   H  BB  HBP  AB  SF POS  yearID    AB     H   BB  HBP   SF  \
0  lawrebr01  10   3    1  53   0   P  1954.0  53.0  10.0  3.0  1.0  0.0   
1  lawrebr01  10   3    1  53   0   P  1954.0  53.0  10.0  3.0  1.0  0.0   
2  lawrebr01  10   3    1  53   0   P  1954.0  53.0  10.0  3.0  1.0  0.0   
3  lawrebr01  10   3    1  53   0   P  1954.0  53.0  10.0  3.0  1.0  0.0   
4  lawrebr01  10   3    1  53   0   P  1954.0  53.0  10.0  3.0  1.0  0.0   

     salary  
0  500000.0  
1  500000.0  
2  500000.0  
3  500000.0  
4  500000.0

In [25]:
len(cleaned_data)

7781783

In [26]:
amalgamated_data = cleaned_data

In [27]:
amalgamated_data.head()

playerID   H  BB  HBP  AB  SF POS  yearID    AB     H   BB  HBP   SF  \
0  lawrebr01  10   3    1  53   0   P  1954.0  53.0  10.0  3.0  1.0  0.0   
1  lawrebr01  10   3    1  53   0   P  1954.0  53.0  10.0  3.0  1.0  0.0   
2  lawrebr01  10   3    1  53   0   P  1954.0  53.0  10.0  3.0  1.0  0.0   
3  lawrebr01  10   3    1  53   0   P  1954.0  53.0  10.0  3.0  1.0  0.0   
4  lawrebr01  10   3    1  53   0   P  1954.0  53.0  10.0  3.0  1.0  0.0   

     salary  
0  500000.0  
1  500000.0  
2  500000.0  
3  500000.0  
4  500000.0

In [28]:
top_list = ['H', 'BB', 'HBP']

In [29]:
bottom_list = ['AB', 'BB' , 'HBP', "SF"]

In [30]:
amalgamated_data['OBP'] = amalgamated_data[top_list].sum(axis=1) / amalgamated_data[bottom_list].sum(axis=1)

In [31]:
amalgamated_data.fillna(0)

playerID    H  BB  HBP   AB  SF POS  yearID     AB      H    BB  \
0        lawrebr01   10   3    1   53   0   P  1954.0   53.0   10.0   3.0   
1        lawrebr01   10   3    1   53   0   P  1954.0   53.0   10.0   3.0   
2        lawrebr01   10   3    1   53   0   P  1954.0   53.0   10.0   3.0   
3        lawrebr01   10   3    1   53   0   P  1954.0   53.0   10.0   3.0   
4        lawrebr01   10   3    1   53   0   P  1954.0   53.0   10.0   3.0   
5        lawrebr01   10   3    1   53   0   P  1954.0   53.0   10.0   3.0   
6        lawrebr01   10   3    1   53   0   P  1954.0   53.0   10.0   3.0   
7        lawrebr01    2   1    0   21   0   P  1955.0   21.0    2.0   1.0   
8        lawrebr01    2   1    0   21   0   P  1955.0   21.0    2.0   1.0   
9        lawrebr01    2   1    0   21   0   P  1955.0   21.0    2.0   1.0   
10       lawrebr01    2   1    0   21   0   P  1955.0   21.0    2.0   1.0   
11       lawrebr01    2   1    0   21   0   P  1955.0   21.0    2.0   1.0   
12       lawrebr01    2   1    0   21   0   P  1955.0   21.0    2.0   1.0   
13       lawrebr01    2   1    0   21   0   P  1955.0   21.0    2.0   1.0   
14       lawrebr01   11   2    1   70   2   P  1956.0   70.0   11.0   2.0   
15       lawrebr01   11   2    1   70   2   P  1956.0   70.0   11.0   2.0   
16       lawrebr01   11   2    1   70   2   P  1956.0   70.0   11.0   2.0   
17       lawrebr01   11   2    1   70   2   P  1956.0   70.0   11.0   2.0   
18       lawrebr01   11   2    1   70   2   P  1956.0   70.0   11.0   2.0   
19       lawrebr01   11   2    1   70   2   P  1956.0   70.0   11.0   2.0   
20       lawrebr01   11   2    1   70   2   P  1956.0   70.0   11.0   2.0   
21       lawrebr01   14   6    0   82   0   P  1957.0   82.0   14.0   6.0   
22       lawrebr01   14   6    0   82   0   P  1957.0   82.0   14.0   6.0   
23       lawrebr01   14   6    0   82   0   P  1957.0   82.0   14.0   6.0   
24       lawrebr01   14   6    0   82   0   P  1957.0   82.0   14.0   6.0   
25       lawrebr01   14   6    0   82   0   P  1957.0   82.0   14.0   6.0   
26       lawrebr01   14   6    0   82   0   P  1957.0   82.0   14.0   6.0   
27       lawrebr01   14   6    0   82   0   P  1957.0   82.0   14.0   6.0   
28       lawrebr01    6   5    0   53   2   P  1958.0   53.0    6.0   5.0   
29       lawrebr01    6   5    0   53   2   P  1958.0   53.0    6.0   5.0   
...            ...  ...  ..  ...  ...  ..  ..     ...    ...    ...   ...   
7781753  taylomi02  108  35    1  472   2  RF  2015.0  472.0  108.0  35.0   
7781754  taylomi02  108  35    1  472   2  LF  2015.0  472.0  108.0  35.0   
7781755  taylomi02  108  35    1  472   2  CF  2015.0  472.0  108.0  35.0   
7781756  taylomi02  108  35    1  472   2  RF  2015.0  472.0  108.0  35.0   
7781757  taylomi02  108  35    1  472   2  OF  2015.0  472.0  108.0  35.0   
7781758  treinbl01    1   1    0   12   0   P  2014.0   12.0    1.0   1.0   
7781759  treinbl01    1   1    0   12   0   P  2014.0   12.0    1.0   1.0   
7781760  treinbl01    0   0    0    2   0   P  2015.0    2.0    0.0   0.0   
7781761  treinbl01    0   0    0    2   0   P  2015.0    2.0    0.0   0.0   
7781762  vargake01   59  12    3  215   4  1B  2014.0  215.0   59.0  12.0   
7781763  vargake01   59  12    3  215   4  1B  2014.0  215.0   59.0  12.0   
7781764  vargake01   42   9    0  175   0  1B  2015.0  175.0   42.0   9.0   
7781765  vargake01   42   9    0  175   0  1B  2015.0  175.0   42.0   9.0   
7781766   wadats01    1   1    0   20   0   P  2014.0   20.0    1.0   1.0   
7781767   wadats01    1   1    0   20   0   P  2014.0   20.0    1.0   1.0   
7781768   wadats01    0   1    0    8   0   P  2015.0    8.0    0.0   1.0   
7781769   wadats01    0   1    0    8   0   P  2015.0    8.0    0.0   1.0   
7781770  yateski01    0   0    0    0   0   P  2014.0    0.0    0.0   0.0   
7781771  yateski01    0   0    0    0   0   P  2014.0    0.0    0.0   0.0   
7781772  yateski01    0   0    0    0   0   P  2015.0    0.0    0.0   0.0   
778177

In [32]:
# def get_obp(hits, walks, hit_by_pitch, at_bats, sacrifice_flies):
#     return ((hits + walks + hit_by_pitch)/(at_bats + walks + hit_by_pitch + sacrifice_flies))

In [33]:
# amalgamated_data['OBP'] = get_obp(cleaned_data['H'], cleaned_data['BB'], cleaned_data['HBP'], cleaned_data['AB'], cleaned_data['SF'])

In [34]:
amalgamated_data = amalgamated_data.fillna(0)

In [35]:
amalgamated_data.head()

playerID   H  BB  HBP  AB  SF POS  yearID    AB     H   BB  HBP   SF  \
0  lawrebr01  10   3    1  53   0   P  1954.0  53.0  10.0  3.0  1.0  0.0   
1  lawrebr01  10   3    1  53   0   P  1954.0  53.0  10.0  3.0  1.0  0.0   
2  lawrebr01  10   3    1  53   0   P  1954.0  53.0  10.0  3.0  1.0  0.0   
3  lawrebr01  10   3    1  53   0   P  1954.0  53.0  10.0  3.0  1.0  0.0   
4  lawrebr01  10   3    1  53   0   P  1954.0  53.0  10.0  3.0  1.0  0.0   

     salary       OBP  
0  500000.0  0.245614  
1  500000.0  0.245614  
2  500000.0  0.245614  
3  500000.0  0.245614  
4  500000.0  0.245614

In [36]:
amalgamated_data = amalgamated_data.drop(amalgamated_data[amalgamated_data.OBP == 1].index)

In [37]:
amalgamated_data.head()

playerID   H  BB  HBP  AB  SF POS  yearID    AB     H   BB  HBP   SF  \
0  lawrebr01  10   3    1  53   0   P  1954.0  53.0  10.0  3.0  1.0  0.0   
1  lawrebr01  10   3    1  53   0   P  1954.0  53.0  10.0  3.0  1.0  0.0   
2  lawrebr01  10   3    1  53   0   P  1954.0  53.0  10.0  3.0  1.0  0.0   
3  lawrebr01  10   3    1  53   0   P  1954.0  53.0  10.0  3.0  1.0  0.0   
4  lawrebr01  10   3    1  53   0   P  1954.0  53.0  10.0  3.0  1.0  0.0   

     salary       OBP  
0  500000.0  0.245614  
1  500000.0  0.245614  
2  500000.0  0.245614  
3  500000.0  0.245614  
4  500000.0  0.245614

In [38]:
len(amalgamated_data)

7757821

In [39]:
amalgamated_data = amalgamated_data.drop(amalgamated_data[amalgamated_data.OBP == 0].index)

In [40]:
amalgamated_data.head()

playerID   H  BB  HBP  AB  SF POS  yearID    AB     H   BB  HBP   SF  \
0  lawrebr01  10   3    1  53   0   P  1954.0  53.0  10.0  3.0  1.0  0.0   
1  lawrebr01  10   3    1  53   0   P  1954.0  53.0  10.0  3.0  1.0  0.0   
2  lawrebr01  10   3    1  53   0   P  1954.0  53.0  10.0  3.0  1.0  0.0   
3  lawrebr01  10   3    1  53   0   P  1954.0  53.0  10.0  3.0  1.0  0.0   
4  lawrebr01  10   3    1  53   0   P  1954.0  53.0  10.0  3.0  1.0  0.0   

     salary       OBP  
0  500000.0  0.245614  
1  500000.0  0.245614  
2  500000.0  0.245614  
3  500000.0  0.245614  
4  500000.0  0.245614

In [41]:
len(amalgamated_data)

6526892

In [42]:
amalgamated_data.head()

playerID   H  BB  HBP  AB  SF POS  yearID    AB     H   BB  HBP   SF  \
0  lawrebr01  10   3    1  53   0   P  1954.0  53.0  10.0  3.0  1.0  0.0   
1  lawrebr01  10   3    1  53   0   P  1954.0  53.0  10.0  3.0  1.0  0.0   
2  lawrebr01  10   3    1  53   0   P  1954.0  53.0  10.0  3.0  1.0  0.0   
3  lawrebr01  10   3    1  53   0   P  1954.0  53.0  10.0  3.0  1.0  0.0   
4  lawrebr01  10   3    1  53   0   P  1954.0  53.0  10.0  3.0  1.0  0.0   

     salary       OBP  
0  500000.0  0.245614  
1  500000.0  0.245614  
2  500000.0  0.245614  
3  500000.0  0.245614  
4  500000.0  0.245614

In [43]:
amalgamated_data = amalgamated_data.drop(amalgamated_data[amalgamated_data.yearID < 2014].index)

In [44]:
amalgamated_data = amalgamated_data.drop_duplicates('playerID')

In [45]:
len(amalgamated_data)

740

In [46]:
amalgamated_data.head()

playerID    H  BB  HBP   AB  SF POS  yearID     AB      H    BB  \
5045722  rodrial01  131  84    6  523   7  SS  2015.0  523.0  131.0  84.0   
5207223  giambja01    8   9    1   60   0  1B  2014.0   60.0    8.0   9.0   
5272287  jeterde01  149  35    6  581   4  SS  2014.0  581.0  149.0  35.0   
5486215  abreubo01   33  20    0  133   2  LF  2014.0  133.0   33.0  20.0   
5614831  ibanera01   26  23    0  166   1  DH  2014.0  166.0   26.0  23.0   

         HBP   SF    salary       OBP  
5045722  6.0  7.0  442333.0  0.356452  
5207223  1.0  0.0  109000.0  0.257143  
5272287  6.0  4.0  130000.0  0.303514  
5486215  0.0  2.0  150000.0  0.341935  
5614831  0.0  1.0  150000.0  0.257895

In [47]:
type(amalgamated_data['OBP'])

pandas.core.series.Series

have list of players, now need to determine OBP per salary, order by that value from best ratio to worst, then find each position from that list.

In [55]:
amalgamated_data['OBP_per_dollar'] = amalgamated_data['OBP'] / amalgamated_data['salary']

In [56]:
amalgamated_data.head()

playerID    H  BB  HBP   AB  SF POS  yearID     AB      H    BB  \
5045722  rodrial01  131  84    6  523   7  SS  2015.0  523.0  131.0  84.0   
5207223  giambja01    8   9    1   60   0  1B  2014.0   60.0    8.0   9.0   
5272287  jeterde01  149  35    6  581   4  SS  2014.0  581.0  149.0  35.0   
5486215  abreubo01   33  20    0  133   2  LF  2014.0  133.0   33.0  20.0   
5614831  ibanera01   26  23    0  166   1  DH  2014.0  166.0   26.0  23.0   

         HBP   SF    salary       OBP            0  OBP_per_dollar  
5045722  6.0  7.0  442333.0  0.356452  8.05844e-07    8.058445e-07  
5207223  1.0  0.0  109000.0  0.257143  2.35911e-06    2.359109e-06  
5272287  6.0  4.0  130000.0  0.303514  2.33473e-06    2.334726e-06  
5486215  0.0  2.0  150000.0  0.341935  2.27957e-06    2.279570e-06  
5614831  0.0  1.0  150000.0  0.257895   1.7193e-06    1.719298e-06

In [57]:
amalgamated_data.sort_values(by='OBP_per_dollar', ascending=False)

playerID    H   BB  HBP   AB   SF  POS  yearID     AB      H  \
5207223         giambja01    8    9    1   60    0   1B  2014.0   60.0    8.0   
5272287         jeterde01  149   35    6  581    4   SS  2014.0  581.0  149.0   
5994837         beltrad01  178   57    3  549    5   3B  2014.0  549.0  178.0   
5486215         abreubo01   33   20    0  133    2   LF  2014.0  133.0   33.0   
7058389         cabremi01  191   60    3  611   11   3B  2014.0  611.0  191.0   
5925289         ortizda01  136   75    3  518    6   1B  2014.0  518.0  136.0   
6956363         lewisco01    2    0    0    5    0    P  2014.0    5.0    2.0   
7632209         burriem01    2    2    0    3    0   2B  2015.0    3.0    2.0   
6227737         ramirar01  141   21   13  494    3   3B  2014.0  494.0  141.0   
6033165         chaveer01   17   11    0   69    1   3B  2014.0   69.0   17.0   
5614831         ibanera01   26   23    0  166    1   DH  2014.0  166.0   26.0   
6606385         pennybr01    2    0    0    6    0    P  2014.0    6.0    2.0   
6771676         pujolal01  172   48    5  633    9   1B  2014.0  633.0  172.0   
6620616         rolliji01  131   64    1  538    3   SS  2014.0  538.0  131.0   
5848096         hunteto01  157   23    7  549    7   CF  2014.0  549.0  157.0   
7243468         kazmisc01    1    0    0    2    0    P  2015.0    2.0    1.0   
7357887         feldmsc01    1    0    0    2    0    P  2015.0    2.0    1.0   
6827405         uribeju01  120   15    1  386    2   SS  2014.0  386.0  120.0   
6009471         beltrca01   94   37    4  403    5   CF  2014.0  403.0   94.0   
5863104         konerpa01   43   10    4  208    2   1B  2014.0  208.0   43.0   
6782434         puntoni01   41   25    0  198    0   SS  2014.0  198.0   41.0   
6691241          dunnad01   80   65    3  363    4   LF  2014.0  363.0   80.0   
6207630         pierzaj01   65    9    4  256    4    C  2014.0  256.0   65.0   
7771035         familje01    2    0    0    3    0    P  2014.0    3.0    2.0   
7179359         bautijo02  158  104    9  553    6   3B  2014.0  553.0  158.0   
6967390         martivi01  188   70    4  561    6    C  2014.0  561.0  188.0   
7028619         werthja01  156   83    9  534    3   CF  2014.0  534.0  156.0   
7781427         norrida01    1    1    0    2    0    P  2015.0    2.0    1.0   
6097456         gonzaal02    5    2    0   30    0   SS  2014.0   30.0    5.0   
7557247         buchhcl01    1    0    0    2    0    P  2014.0    2.0    1.0   
...                   ...  ...  ...  ...  ...  ...  ...     ...    ...    ...   
7781269         foltymi01    2    0    0   28    1    P  2015.0   28.0    2.0   
7773466         iwakuhi01    0    1    0    4    0    P  2014.0    4.0    0.0   
7768786         turneja01    3    1    0   13    0    P  2014.0   13.0    3.0   
7750685         vicieda01  121   32    5  523    3   1B  2014.0  523.0  121.0   
7781652         solerjo01   26    6    0   89    2   OF  2014.0   89.0   26.0   
7772852         hechaad01  148   26    1  536    6   2B  2014.0  536.0  148.0   
7781128         anderch01    1    0    1   34    0    P  2014.0   34.0    1.0   
7779112          haleda02    1    0    0   17    1    P  2014.0   17.0    1.0   
7759096         fiersmi01    1    0    0   19    0    P  2014.0   19.0    1.0   
7763494         martile01  146   39    2  533    2   CF  2014.0  533.0  146.0   
6648599         beckejo02    7    1    0   39    0    P  2014.0   39.0    7.0   
7781172         butleed01    1    0    0   23    0    P  2015.0   23.0    1.0   
7781236         despaod01    0    1    0   24    0    P  2014.0   24.0    0.0   
7760588          handbr01    1    0    0   27    0    P  2014.0   27.0    1.0   
7780828         wachami01    1    0    0   29    0    P  2014.0   29.0    1.0   
6815978         suzukic01  102   21    1  359    2   OF  2014.0  359.0  102.0   
7671862         samarje01    6    0    0   34    2    P  2014.0   34.0    6.0   
7781778         tomasya01  111   17    2  406   

In [59]:
final_data_set = amalgamated_data[['playerID', 'POS', 'yearID', 'salary', 'OBP', 'OBP_per_dollar']]

In [64]:
final_data_set.sort_values(by='OBP_per_dollar', ascending=False)

playerID  POS  yearID      salary       OBP  OBP_per_dollar
5207223         giambja01   1B  2014.0    109000.0  0.257143    2.359109e-06
5272287         jeterde01   SS  2014.0    130000.0  0.303514    2.334726e-06
5994837         beltrad01   3B  2014.0    170000.0  0.387622    2.280130e-06
5486215         abreubo01   LF  2014.0    150000.0  0.341935    2.279570e-06
7058389         cabremi01   3B  2014.0    165574.0  0.370803    2.239500e-06
5925289         ortizda01   1B  2014.0    170000.0  0.355482    2.091069e-06
6956363         lewisco01    P  2014.0    200000.0  0.400000    2.000000e-06
7632209         burriem01   2B  2015.0    404000.0  0.800000    1.980198e-06
6227737         ramirar01   3B  2014.0    170000.0  0.329567    1.938629e-06
6033165         chaveer01   3B  2014.0    201000.0  0.345679    1.719796e-06
5614831         ibanera01   DH  2014.0    150000.0  0.257895    1.719298e-06
6606385         pennybr01    P  2014.0    200000.0  0.333333    1.666667e-06
6771676         pujolal01   1B  2014.0    200000.0  0.323741    1.618705e-06
6620616         rolliji01   SS  2014.0    200000.0  0.323432    1.617162e-06
5848096         hunteto01   CF  2014.0    200000.0  0.319113    1.595563e-06
7243468         kazmisc01    P  2015.0    316000.0  0.500000    1.582278e-06
7357887         feldmsc01    P  2015.0    329000.0  0.500000    1.519757e-06
6827405         uribeju01   SS  2014.0    222000.0  0.336634    1.516368e-06
6009471         beltrca01   CF  2014.0    200000.0  0.300668    1.503341e-06
5863104         konerpa01   1B  2014.0    170000.0  0.254464    1.496849e-06
6782434         puntoni01   SS  2014.0    200000.0  0.295964    1.479821e-06
6691241          dunnad01   LF  2014.0    250000.0  0.340230    1.360920e-06
6207630         pierzaj01    C  2014.0    210000.0  0.285714    1.360544e-06
7771035         familje01    P  2014.0    491750.0  0.666667    1.355702e-06
7179359         bautijo02   3B  2014.0    300000.0  0.403274    1.344246e-06
6967390         martivi01    C  2014.0    304500.0  0.408736    1.342320e-06
7028619         werthja01   CF  2014.0    300000.0  0.394277    1.314255e-06
7781427         norrida01    P  2015.0    508700.0  0.666667    1.310530e-06
6097456         gonzaal02   SS  2014.0    170000.0  0.218750    1.286765e-06
7557247         buchhcl01    P  2014.0    396000.0  0.500000    1.262626e-06
...                   ...  ...     ...         ...       ...             ...
7781269         foltymi01    P  2015.0    508750.0  0.068966    1.355588e-07
7773466         iwakuhi01    P  2014.0   1500000.0  0.200000    1.333333e-07
7768786         turneja01    P  2014.0   2150000.0  0.285714    1.328904e-07
7750685         vicieda01   1B  2014.0   2250000.0  0.280639    1.247286e-07
7781652         solerjo01   OF  2014.0   2666667.0  0.329897    1.237113e-07
7772852         hechaad01   2B  2014.0   2750000.0  0.307557    1.118390e-07
7781128         anderch01    P  2014.0    512500.0  0.057143    1.114983e-07
7779112          haleda02    P  2014.0    502500.0  0.055556    1.105583e-07
7759096         fiersmi01    P  2014.0    498000.0  0.052632    1.056859e-07
7763494         martile01   CF  2014.0   3250000.0  0.324653    9.989316e-08
6648599         beckejo02    P  2014.0   2156250.0  0.200000    9.275362e-08
7781172         butleed01    P  2015.0    509500.0  0.043478    8.533515e-08
7781236         despaod01    P  2014.0    517300.0  0.040000    7.732457e-08
7760588          handbr01    P  2014.0    500000.0  0.037037    7.407407e-08
7780828         wachami01    P  2014.0    510000.0  0.034483    6.761325e-08
6815978         suzukic01   OF  2014.0   5666667.0  0.323760    5.713408e-08
7671862         samarje01    P  2014.0   3000000.0  0.166667    5.555556e-08
7781778         tomasya01   1B  2015.0   5500000.0  0.305164    5.548442e-08
7781096         abreujo02   1B  2014.0   7000000.0  0.382637    5.466238e-08
7780568           ryuhy01    P  2014.0   3333333.0  0.166667    5.000001e-08
7747433      

# Starting Line-Up
1. lewisco01	P	2014.0	200000.0	0.400000	2.000000e-06
2. pierzaj01	C	2014.0	210000.0	0.285714	1.360544e-06
3. giambja01	1B	2014.0	109000.0	0.257143	2.359109e-06
4. burriem01	2B	2015.0	404000.0	0.800000	1.980198e-06
5. jeterde01	SS	2014.0	130000.0	0.303514	2.334726e-06
6. beltrad01	3B	2014.0	170000.0	0.387622	2.280130e-06
7. abreubo01	LF	2014.0	150000.0	0.341935	2.279570e-06
8. hunteto01	CF	2014.0	200000.0	0.319113	1.595563e-06
9. solerjo01	OF	2014.0	2666667.0	0.329897	1.237113e-07